mbtools/models.py

```diff
@@ -120,39 +141,74 @@ def delete_empty_categories(self):
+        self.id = int(self.tree.attrib['id'])
+
+    def name(self):
+        return self.tree.xpath("title")[0].text
+
+    def next(self):
+        return int(self.tree.xpath("nextpageid")[0].text)
+
+    def prev(self):
+        return int(self.tree.xpath("prevpageid")[0].text)
+
+    def answers(self):
+        answer_objs = []
+        for answer in self.tree.xpath("answers/answer"):
+            answer_objs.append(
+                MoodleLessonAnswer(answer, self.location)
+            )
+        return answer_objs
+
+    def html_element(self):
+        return MoodleHtmlElement(self.tree.xpath("contents")[0], self.location)
+
+    def html_elements(self):
+        elems = []
+        elems.append(self.html_element())
+        for answer in self.answers():
+            elems.append(answer.html_element())
+        return elems
+
+
+class MoodleLessonAnswer:
+    def __init__(self, tree, location):
+        self.tree = tree
+        self.location = location + f' (answer id: {self.tree.attrib["id"]})'
+
+    def name(self):
+        return self.tree.attrib["id"]
+
+    def html_element(self):
+        answer_text = self.tree.xpath("answer_text")[0]
+        return MoodleHtmlElement(answer_text, self.location)
+
+
 class MoodleLesson:
     def __init__(self, activity_path, mbz_path):
         self.mbz_path = Path(mbz_path)
         self.activity_path = activity_path
         self.activity_filename = str(self.activity_path / "lesson.xml")
         self.etree = etree.parse(self.activity_filename)
 
+    def name(self):
+        return self.etree.xpath("//name")[0].text
+
+    def lesson_pages(self):
+        page_objs = []
+        for page in self.etree.xpath("//pages/page"):
+            page_objs.append(MoodleLessonPage(page, self.name()))
+        return page_objs
+
     def html_elements(self):
-        elements = []
-        lesson_name = self.etree.xpath("//name")[0].text
-        pages = self.etree.xpath("//pages/page")
-        for page in pages:
-            page_id = page.attrib["id"]
-            page_title = page.xpath("title")[0].text
-            contents = page.xpath("contents")
-
-            location = \
-                f'Lesson: {lesson_name} (page id={page_id}) ' \
-                f'Page Title: {page_title}'
-            elements.append(MoodleHtmlElement(contents[0], location))
-
-            for answer in page.xpath("answers/answer"):
-                answer_format = answer.xpath("answerformat")[0]
-                # We need to check the answer format to filter out things
-                # like buttons which are also serialized as answers but are
-                # not HTML
-                if answer_format.text == "1":
-                    answer_text = answer.xpath("answer_text")[0]
-                    answer_location = \
-                        f'{location} (answer id: {answer.attrib["id"]})'
-                    elements.append(
-                        MoodleHtmlElement(answer_text, answer_location))
```

Comment:
@rnathuji I am not sure what you mean by test_models.py creating a `MoodleLesson` instance without the builders... while building here I still feel like the easiest way to test the answers created is just to use a lesson builder and an mbz_builder...

Reply:
@mjoss17 - Whoops, first off I meant `MoodleLessonPage` instead of `MoodleLesson` in my comment.

So the general idea is we need a test where there's content that includes an answer with `<answerformat>0</answerformat>` and validate that it's `<answer_text>` is not returned as a `MoodleHtmlElement`. If you try to use the builder fixtures to write that test, the complicating factor is the fixtures currently always set `<answerformat>1</answerformat>` per my link above. So to use the builder for this one test case we'd have to make a change to make that configurable in the fixtures which seems overkill to me. Instead, if it were me, I'd write a really simple test case that avoid that. I'll try to draft up an example of what I'm thinking tomorrow to help clear things up.

Reply:
Actually, I just whipped an example together real quick now since it was in my head :sweat_smile: : 

```python
def test_answerformat_filter(tmp_path):
    lesson_page_xml = f"""
<lesson>
  <name>Lesson name</name>
  <pages>
    <page id="0">
      <contents>Lesson content</contents>
      <title>Lesson page title</title>
      <answers>
        <answer id="1">
          <answerformat>0</answerformat>
          <answer_text>Not HTML answer content</answer_text>
        </answer>
        <answer id="2">
          <answerformat>1</answerformat>
          <answer_text>HTML answer content</answer_text>
        </answer>
      </answers>
    </page>
  </pages>
</lesson>
  """

    lesson_page_etree = etree.fromstring(lesson_page_xml).xpath("//page")[0]
    lesson_page = MoodleLessonPage(lesson_page_etree)
    html_elems = lesson_page.html_elements()
    assert len(html_elems) == 2
    assert "Not HTML answer content" not in [elem.parent.text for elem in html_elems]
```

This test will fail currently due to the code you removed.

Reply:
I see, thank you! - yes going the other way with the builders and hacking together a way to change `answerformats` was looking ugly when I really got down to it 😅.